lib/iris/_concatenate.py

```diff
@@ -464,6 +465,12 @@ def match(self, other, error_on_mismatch):
             msgs.append(msg_template.format('Datatypes', '',
                                             self.data_type, other.data_type))
 
+        # Check _cell_measures_and_dims
+        if self._cell_measures_and_dims != other._cell_measures_and_dims:
```

Comment:
> Does the order of cell measures within cell_measures_and_dims matter?

this is a tricky one.  I can see how we could decide that a reordered list is equivalent, but it would require more code to test for.  This is very much an edge case, it seems unlikely to occur.

do you want me to explicitly implement an unordered test?
if so, how might you suggest?